src/t8_cmesh/t8_cmesh_vtk_reader.cxx

```diff
@@ -0,0 +1,101 @@
+/*
+This file is part of t8code.
+t8code is a C library to manage a collection (a forest) of multiple
+connected adaptive space-trees of general element classes in parallel.
+
+Copyright (C) 2015 the developers
+
+t8code is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+t8code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with t8code; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_cmesh_vtk_reader.cxx
+* Implementation of a Reader for vtk/vtu files using the vtk-library.
+* The functions can only be used when t8code is linked with the vtk-library.
+*/
+
+#include <t8_cmesh.h>
+#include <t8_cmesh_vtk_writer.h>
+#include <t8_cmesh_vtk_reader.hxx>
+#include <t8_cmesh/t8_cmesh_examples.h>
+#include <t8_cmesh/t8_cmesh_vtk_helper.hxx>
+
+#if T8_WITH_VTK
+#include <vtkUnstructuredGrid.h>
+#include <vtkAbstractArray.h>
+#include <vtkTriangleFilter.h>
+#include <vtkCellData.h>
+#include <vtkPolyData.h>
+#include <vtkTriangleFilter.h>
+#endif
+T8_EXTERN_C_BEGIN ();
+
+/*Construct a cmesh given a filename and a*/
```

Comment:
and a what?!

Now i am hooked and want to know what comes next...please tell me!

Reply:
Gonna tell you in season two of "You won't believe what happened next"